### PR TITLE
log handler deprecated filename_template argument removal

### PR DESCRIPTION
**Pull request by dirrao**
Passing filename_template to a log handler is deprecated and has no effect. So, removing deprecated filename_template argument in log handler.

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

File: airflow/providers/microsoft/azure/log/wasb_task_handler.py [hunk 1]
```diff
@@ -46,14 +46,13 @@ class WasbTaskHandler(FileTaskHandler, LoggingMixin):
 
     def __init__(
         self,
+        *,
```
Comment (romsharon98) [inline airflow/providers/microsoft/azure/log/wasb_task_handler.py]: I think moving `*` to this place is a breaking change.

Comment (dirrao) [reply airflow/providers/microsoft/azure/log/wasb_task_handler.py]: I have removed. It's not needed anymore.

**Review by romsharon98 — APPROVED** [review]

**Comment by eladkal** [comment]
This PR involves both core and provider changes.
Please clarify what it means for new versions of providers (that will include this PR) but installed with Airflow 2.
Something is odd here.

The parameter in question is public API which is why we deprecate it first. Now main is Airflow 3 so we can remove it but providers must still be compatible with Airflow 2... so how can we remove it from providers safely now?

**Comment by dirrao** [comment]
> This PR involves both core and provider changes. Please clarify what it means for new versions of providers (that will include this PR) but installed with Airflow 2. Something is odd here.
> 
 Right now, this field is not being used and has no effect. The filename_template is optional argument with default value None.
 We can still able to install and use latest providers on airflow 2 without any issue. 

> The parameter in question is public API which is why we deprecate it first. Now main is Airflow 3 so we can remove it but providers must still be compatible with Airflow 2... so how can we remove it from providers safely now?

As I said in the above statement, the providers can still work with airflow 2.

**Comment by potiuk** [comment]
@eladkal is right - if someone  installs the new provider witn an old configuration in Airlfow 2 with filename_template, it will just fail with "attemptign to iniitalize handlew with "filename_template" argument passed.

**Comment by dirrao** [comment]
I see the keyword arguments for the all the 

> @eladkal is right - if someone installs the new provider witn an old configuration in Airlfow 2 with filename_template, it will just fail with "attemptign to iniitalize handlew with "filename_template" argument passed but it's missing" (or smth like that).

Yah. You are right. Most of them are supporting keyword arguments except one/two  (like. es handler).

**Comment by dirrao** [comment]
@eladkal / @potiuk 
Let me know if you want me to revert it for providers alone.

**Comment by potiuk** [comment]
Just reverting will not be enough, but I think if the base Handler will accept extra arguments (and ignores them) we should be home

**Pull request merged**